From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jeff Genovy <29107334+jefgen@users.noreply.github.com>
Date: Tue, 8 Dec 2020 12:07:27 -0800
Subject: MSFT-PATCH: CLDR-to-ICU data tool changes for MS-ICU/Windows (Part 2)

Windows only searches for .exe executables by default, so we can't just
call "ant" and instead need to launch a cmd prompt to run it.

The "&quot;" changes are needed in order double-quote the arguments
that get passed to the JVM. Without this hack the arguments don't get
quoted at all, which causes build failures as some of the variables
which get replaced have embedded spaces in them.

diff --git a/tools/cldr/cldr-to-icu/build-icu-data.xml b/tools/cldr/cldr-to-icu/build-icu-data.xml
index d0cebe8ab2..5293a10147 100644
--- a/tools/cldr/cldr-to-icu/build-icu-data.xml
+++ b/tools/cldr/cldr-to-icu/build-icu-data.xml
@@ -123,27 +123,37 @@
          need to use CLDR_DIR at runtime to find the production data, this can all be
          removed. -->
     <target name="convert" depends="init-args, prepare-jar">
-        <exec executable="ant" searchpath="true" failonerror="true">
+        <!-- MSFT-Change: Windows only searches for .exe executables by default, so we can't just
+             call "ant" here, and instead need to launch a cmd prompt to run it.
+             
+             The "&quot;" below is needed in order double-quote the arguments that get passed to
+             the JVM. Without this hack the arguments don't get quoted at all, which causes build
+             failures as some of the variables which get replaced below have embedded spaces in them.
+         -->
+        <exec executable="cmd.exe" searchpath="true" failonerror="true">
+            <arg value="/c"/>
+            <arg value="ant.bat"/>
             <!-- The CLDR library wants CLDR_DIR set, to the data directory. -->
             <env key="CLDR_DIR" value="${cldrDataDir}" />
             <!-- Force inherit JAVA_HOME (this can be important). -->
             <env key="JAVA_HOME" value="${javaHome}" />
             <!-- Initial Ant command line with all the "interesting" bit in. -->
-            <arg line="-f build-icu-data.xml convert-impl -DcldrDir=${cldrDataDir}"/>
+            <arg line="-f build-icu-data.xml convert-impl"/>
+            <arg value="&quot;-DcldrDir=${cldrDataDir}&quot;"/>
             <!-- List all properties in the "convert-impl" task (except cldrDir). -->
-            <arg value="-DoutDir=${outDir}"/>
-            <arg value="-DgenCCodeDir=${genCCodeDir}"/>
-            <arg value="-DgenJavaCodeDir=${genJavaCodeDir}"/>
-            <arg value="-DdontGenCode=${dontGenCode}"/>
-            <arg value="-DspecialsDir=${specialsDir}"/>
-            <arg value="-DoutputTypes=${outputTypes}"/>
-            <arg value="-DicuVersion=${icuVersion}"/>
-            <arg value="-DicuDataVersion=${icuDataVersion}"/>
-            <arg value="-DcldrVersion=${cldrVersion}"/>
-            <arg value="-DminDraftStatus=${minDraftStatus}"/>
-            <arg value="-DlocaleIdFilter=${localeIdFilter}"/>
-            <arg value="-DincludePseudoLocales=${includePseudoLocales}"/>
-            <arg value="-DemitReport=${emitReport}"/>
+            <arg value="&quot;-DoutDir=${outDir}&quot;"/>
+            <arg value="&quot;-DgenCCodeDir=${genCCodeDir}&quot;"/>
+            <arg value="&quot;-DgenJavaCodeDir=${genJavaCodeDir}&quot;"/>
+            <arg value="&quot;-DdontGenCode=${dontGenCode}&quot;"/>
+            <arg value="&quot;-DspecialsDir=${specialsDir}&quot;"/>
+            <arg value="&quot;-DoutputTypes=${outputTypes}&quot;"/>
+            <arg value="&quot;-DicuVersion=${icuVersion}&quot;"/>
+            <arg value="&quot;-DicuDataVersion=${icuDataVersion}&quot;"/>
+            <arg value="&quot;-DcldrVersion=${cldrVersion}&quot;"/>
+            <arg value="&quot;-DminDraftStatus=${minDraftStatus}&quot;"/>
+            <arg value="&quot;-DlocaleIdFilter=${localeIdFilter}&quot;"/>
+            <arg value="&quot;-DincludePseudoLocales=${includePseudoLocales}&quot;"/>
+            <arg value="&quot;-DemitReport=${emitReport}&quot;"/>
         </exec>
     </target>
 
